pymeasure/instruments/oxfordinstruments/ips120_10.py

```diff
@@ -249,28 +249,31 @@ def switch_heater_enabled(self, value):
         self.switch_heater_status = status_value
 
     current_setpoint = Instrument.control(
-        "R0", "$I%f",
+        "R0", "$I%d",
         """ A floating point property that controls the magnet current set-point of
         the IPS in ampere. """,
-        get_process=lambda v: float(v[1:]),
+        get_process=lambda v: float(v[1:]) * 1e-2,
+        set_process=lambda v: int(v * 1e2),
         validator=truncated_range,
         values=[0, _MAX_CURRENT]
     )
```

Comment:
Sorry for getting back to this so late.

I have to say that I find this conversion from float to int a tad bit strange as the manual of the IPS120-10 mentions explicitly that it treats all numbers as signed decimals (in contrast to the older PS120-10, which treats numbers as signed integers). Especially since the values I got when using the IPS were indeed decimals and didn't need conversion from and to integers. Could you comment on this?

Reply:
Ha, well, you are absolutely right. 
It turns out, that we have in the lab the manual for the IPS, but the device is the PS, no intelligence was delivered to us ;)
Since we also have the ITC 503 and the ILM 211, all "intelligent devices", I did not bother to look what was written on the box, after seeing the manual. Now it is more clear why it behaves "not as described in the manual". 

I propose that I widen the PR, so that we have both the PS and the IPS, the only difference being after all the int/float conversion. What do you think? 
Possibly we could make a base class of, let's say `Oxford_magnet_powersupply_functions`, and inherit this for the classes for the `IPS` and the `PS`, where all the properties are then defined separately? Or can we do this more elegantly, somehow wrapping the respective properties in `get_process` and `set_process` functions?

Reply:
Ahh that makes sense.

Having an class for both the IPS and the PS would be cool indeed. The nicest way would be to use the dynamic property that is currently being implemented #376 (this has been in the pipeline for quite some time, but I think it won't take too long now); in that case we could just base the PS120-10 on the IPS120-10 and change the `set_process`, `get_process`, and `set_command`. However, this is at this moment not yet merged.

What I could suggest (also as it is always nice to limit the size of a PR) to first merge the IPS, wait for the dynamic property PR and then also implement the PS (that would save a bit for work but might take a bit longer)? Would that work for you? Another option would be to temporarily have them both implemented until the dynamic property is merged and then modifying one of the classes (a bit more work).

Reply:
Sure, well, I guess we could get away with having them temporarily implemented in a quick-and-dirty way, and clean it up once the nice tools are available. like this, the outside API for users (like me) won't change, but we can have some improvements under the hood later on. 
In this case however, I would not bother with making it half-way nice now, but would just copy the IPS as the PS, and remove the corresponding things in the `get_process` and so on, so we have quite a lot of copy-pasted code until the PR you mentioned is in, but I will then fix it up quickly, and at the same time get a feel for the new possibilities with the dynamic control stuff. 
I guess I will address this question in your own PR at pymeasure, so others can have a say in that too, because, after all, quick-and-dirty solutions have a tendency to stick, despite all good intentions, and others might disagree with having it like this in the master branch.

Reply:
FYI: https://github.com/pymeasure/pymeasure/pull/376 has landed now!